src/lib/coda_lib/coda_lib.ml

```diff
@@ -572,7 +582,13 @@ let add_work t (work : Snark_worker_lib.Work.Result.t) =
   let spec = work.spec.instances in
   set_seen_jobs t (Work_selection_method.remove (seen_jobs t) spec) ;
   let _ = Or_error.try_with (fun () -> update_metrics ()) in
-  Network_pool.Snark_pool.add_completed_work (snark_pool t) work
+  Strict_pipe.Writer.write t.pipes.local_snark_work_writer
+    (Network_pool.Snark_pool.Resource_pool.Diff.of_result work)
+  |> Deferred.don't_wait_for
+
+let add_transactions t (txns : User_command.t list) =
```

Comment:
I feel like these pipes should be buffered if we're not going to use them synchronously. As is this is essentially a plain pipe. Is waiting for the write to complete a problem?

Reply:
The client could hang for a bit if the diff is not read yet (Local transactions are second in priority after transition frontier diffs.) I could make it buffered and have a capacity of 200 or so and see if that's sufficient during QA?

Reply:
I feel like the client hanging while the transaction is being submitted is the correct behavior.

Reply:
Could you make a TODO and an issue for this one? I agree with Ember here we should probably change it to hang in the future (but for now keeping the don't_wait_for is fine)